src/components/tables/Table.tsx

```diff
@@ -9,25 +9,124 @@
  */
 
 import classNames from 'classnames';
+import { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
 import CommonProps from '../../common/interfaces/common-props.interface';
 
 interface Props extends CommonProps {
   withoutPadding?: boolean;
+  withoutBottomBorder?: boolean;
+  withoutTopBorder?: boolean;
+  withoutLeftBorder?: boolean;
+  withoutRightBorder?: boolean;
+  onVerticalOverflowChange?: (overflow: boolean) => void;
 }
 
 export function Table(props: Props) {
+  const [t] = useTranslation();
+  const { onVerticalOverflowChange } = props;
+
+  const [tableParentHeight, setTableParentHeight] = useState<number>();
+  const [tableHeight, setTableHeight] = useState<number>();
+  const [manualTableHeight, setManualTableHeight] = useState<
+    number | string | undefined
+  >(props.style?.height);
+  const [numberOfCells, setNumberOfCells] = useState<number>(0);
+  const [isVerticallyOverflow, setIsVerticallyOverflow] =
+    useState<boolean>(true);
+
+  const handleTableParentHeight = (element: HTMLDivElement | null) => {
+    if (element && onVerticalOverflowChange) {
+      setTableParentHeight(element.clientHeight);
+    }
+  };
+
+  const handleTableHeight = (element: HTMLTableElement | null) => {
+    if (element && onVerticalOverflowChange) {
+      if (element.querySelectorAll('tbody > tr').length > 0) {
+        if (
+          element.querySelectorAll('tbody > tr > td')[0].textContent ===
+          t('no_records_found')
+        ) {
+          setNumberOfCells(-1);
+        } else {
+          setNumberOfCells(
+            element.querySelectorAll('tbody > tr > td').length ?? 0
+          );
+        }
+      }
+
+      setTableHeight(element.clientHeight);
+    }
+  };
```

Comment:
Oof, 3 levels if. What are we trying to do here? There has to be easier/cleaner way.

Reply:
I changed the logic so I avoided this part with optimized solution.